docs/dev/compiler-toolchain/llvm.md

```diff
@@ -0,0 +1,22 @@
+# LLVM Framework
+
+[The LLVM framework](https://github.com/matter-labs/era-compiler-llvm) is an excellent choice for developing toolchains for smart contract languages. Its powerful intermediate representation (IR) allows developers to design, implement, and optimize efficient
+language-specific features while benefiting from the extensive LLVM ecosystem. This ensures high-performance execution,
+improved portability, and seamless integration with existing LLVM-based tools. Furthermore, the modularity and
+extensibility of LLVM make it easier to support new smart contract languages with LLVM front-ends.
+
+Additionally, LLVM improves on efficiency when compared to the original EVM pipeline, as we can take advantage of the
+numerous optimization passes and tools available in its mature ecosystem.
+
+In our toolchain, LLVM consumes the LLVM IR, applies extensive optimizations, and eventually passes the optimized IR
+to the zkEVM back-end code generator in order to produce the zkEVM text assembly output.
+
+## Optimizer
+
+All our compilers utilize the state-of-the-art LLVM optimizer.
+By default, they optimize for performance, which correlates with the number of VM cycles per transaction,
+thus affecting gas usage. The `z` option may reduce the contract size for large contracts, making deployments cheaper while increasing the average transaction price.
+
```

Comment:
We could add some zkEVM target specific notes:
```
## Things to note

We currently are basing our development on LLVM 15, with the plan to keep the backend up-to-date with the latest (or a recent) upstream release.

Users should note that as of now, our backend now emits text assembly files because the zkEVM assembler and disassembler is not implemented using LLVM framework. Due to historical reasons the assembler is implemented separately, which consumes the assembly emitted by our LLVM backend. We have a plan to implement the assembler, as well as other helpful utilities within LLVM framework soon.

zkEVM architecture uses 256-bit integer registers. To support large integer computation, we expanded the LLVM's integer datatypes with a new data type: `i256` which represents 256bit integers. Here are some important things a potential developer should heed:
* zkEVM target does not support floating types or vector types.
* Smaller datatypes such as `i64` are supported and will be legalized as `i256` internally. In most scenarios, smaller integer types are as fast as 256-bit integer (when extension or truncation are free) but could be slower if alignment restrictions are involved.
* Intrinsic functions: 
    * We introduced a few intrinsic functions which are unique to zkEVM, please see corresponding documents (TODO).
    * Most of the LLVM intrinsic functions are supported, with only a few rare exceptions. 
```

Reply:
@lialan Can you open a new ticket for this in the docs board "triage" section please with these details and we will prioritize accordingly. Thank you.